src/sentry/workflow_engine/processors/workflow.py

```diff
@@ -24,70 +24,51 @@
 WORKFLOW_ENGINE_BUFFER_LIST_KEY = "workflow_engine_delayed_processing_buffer"
 
 
-# TODO remove this method
-def get_data_condition_groups_to_fire(
-    workflows: set[Workflow], job: WorkflowJob
-) -> dict[int, list[int]]:
-    workflow_action_groups: dict[int, list[int]] = defaultdict(list)
+class WorkflowDataConditionGroupType(StrEnum):
+    ACTION_FILTER = "action_filter"
+    WORKFLOW_TRIGGER = "workflow_trigger"
 
-    workflow_ids = {workflow.id for workflow in workflows}
-
-    workflow_dcgs = WorkflowDataConditionGroup.objects.filter(
-        workflow_id__in=workflow_ids
-    ).select_related("condition_group", "workflow")
-
-    for workflow_dcg in workflow_dcgs:
-        action_condition = workflow_dcg.condition_group
-        evaluation, result, _ = evaluate_condition_group(action_condition, job)
-
-        if evaluation:
-            workflow_action_groups[workflow_dcg.workflow_id].append(action_condition.id)
 
-    return workflow_action_groups
-
-
-def enqueue_workflows(
-    workflows: set[Workflow],
-    job: WorkflowJob,
+def enqueue_workflow(
+    workflow: Workflow,
+    delayed_conditions: list[DataCondition],
+    event: GroupEvent,
+    source: WorkflowDataConditionGroupType,
 ) -> None:
-    event = job["event"]
     project_id = event.group.project.id
-    workflow_action_groups = get_data_condition_groups_to_fire(workflows, job)
 
-    for workflow in workflows:
-        buffer.backend.push_to_sorted_set(key=WORKFLOW_ENGINE_BUFFER_LIST_KEY, value=project_id)
-
-        action_filters = workflow_action_groups.get(workflow.id, [])
-        if not action_filters:
-            continue
+    buffer.backend.push_to_sorted_set(key=WORKFLOW_ENGINE_BUFFER_LIST_KEY, value=project_id)
 
-        action_filter_fields = ":".join(map(str, action_filters))
+    condition_groups = ",".join(
+        str(condition.condition_group_id) for condition in delayed_conditions
+    )
```

Comment:
nit: should this be made into a set first to avoid duplicates? (we may or may not need the conditions to collect the DCGs but that can easily be changed)